### PR TITLE
Add virtual role for indexmanagement cronjobs

### PR DESCRIPTION
**Pull request by jcantrill**
This PR fixes IndexManagement crons to:

* utilize SA token
* Create role to a virtual resource for indexmanagent
* Configure proxy to evaluate role

cc @ewolinetz 

depends on https://github.com/openshift/origin-aggregated-logging/pull/1863

### CODE REVIEW

File: pkg/indexmanagement/scripts.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ code=$(curl "$ES_SERVICE/$POLICY_MAPPING-write/_rollover?pretty" \
   --key /etc/indexmanagement/keys/admin-key \
   -HContent-Type:application/json \
   -XPOST \
+  -H"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
```
Comment (ewolinetz) [inline pkg/indexmanagement/scripts.go]: if we're going to use a SA token, i dont think we need to provide the `--cert` and `--key` args

Comment (jcantrill) [reply pkg/indexmanagement/scripts.go]: doesn't the cert/key allow mutual tls at the connection level?  This was the only reason I added it.  not sure if its relevent

Comment (ewolinetz) [reply pkg/indexmanagement/scripts.go]: right, providing the cert and key makes this a mTLS request which I don't think we need to connect to ES (it should be optional)

Comment (jcantrill) [reply pkg/indexmanagement/scripts.go]: @ewolinetz by optional are you suggesting we should allow user's to make it optional or do you believe in this case it is save to remove cert and key?

Comment (ewolinetz) [reply pkg/indexmanagement/scripts.go]: in this case it is safe to remove

**Comment by ewolinetz** [comment]
@jcantrill remove hold when you're ready for this to go in

**Pull request merged**